### PR TITLE
[Typing][A-94] Add type annotations for `python/paddle/signal.py`

### PR DESCRIPTION
**Pull request by Asthestarsfalll**
### PR Category

User Experience
### PR Types

Improvements

### Description

Add type annotations for `python/paddle/signal.py`
- #65008

### CODE REVIEW

File: python/paddle/signal.py [hunk 1]
```diff
@@ -21,13 +24,22 @@
...
     'istft',
 ]
 
 
-def frame(x, frame_length, hop_length, axis=-1, name=None):
+def frame(
+    x: Tensor,
+    frame_length: int,
+    hop_length: int,
+    axis: int = -1,
```
Comment (megemini) [inline python/paddle/signal.py]: Literal ?

File: python/paddle/signal.py [hunk 2]
```diff
@@ -142,7 +156,9 @@ def frame(x, frame_length, hop_length, axis=-1, name=None):
     return out
 
 
-def overlap_add(x, hop_length, axis=-1, name=None):
+def overlap_add(
+    x: Tensor, hop_length: int, axis: int = -1, name: str | None = None
```
Comment (megemini) [inline python/paddle/signal.py]: 同上 ～

File: python/paddle/signal.py (hunk 1 above)
Comment (Asthestarsfalll) [reply python/paddle/signal.py]: 已添加

**Review by megemini — APPROVED** [review]
LGTM ~ 👏

**Review by SigureMo — APPROVED** [review]
LGTMeow

**Pull request merged**